packages/pnpm-sync-lib/src/pnpmSyncPrepare.ts

```diff
@@ -219,10 +224,12 @@ export async function pnpmSyncPrepareAsync(options: IPnpmSyncPrepareOptions): Pr
   const pnpmModulesYaml = YAML.parse(fs.readFileSync(`${pnpmModulesYamlPath}/.modules.yaml`, 'utf8'));
   const pnpmVersion: string | undefined = pnpmModulesYaml?.packageManager?.split('@')[1];
 
-  // currently, only support pnpm v8 and v9
-  if (!pnpmVersion || !(pnpmVersion.startsWith('8') || pnpmVersion.startsWith('9'))) {
+  if (
+    !pnpmVersion ||
+    !(pnpmVersion.startsWith('8') || pnpmVersion.startsWith('9') || pnpmVersion.startsWith('10'))
```

Comment:
Seems like `'80'` also `startsWith('8')`.  Should we check for something like `8.`?

Reply:
You could parse it into a number and then do a numeric comparison for that one.  I doubt that `lockfileVersion` will use `X.Y.Z` syntax -- it is probably just an integer (or maybe regular decimal).

@zkochan